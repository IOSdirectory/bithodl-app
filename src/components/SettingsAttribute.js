import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  container: {
    paddingTop: 12,
    paddingBottom: 12,
    paddingRight: 15,
    marginLeft: 15,
    borderBottomColor: '#DDDDDF',
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  label: {
    fontWeight: '600'
  },
  value: {
    position: 'absolute',
    top: 12,
    right: 15,
    color: '#8F8E94',
    textAlign: 'right',
    width: 150
  }
});

export default class SettingsAttribute extends Component {
  render() {
    const isLastItem = this.props.isLastItem;

    const containerStyles = [
      styles.container,
      isLastItem ? { borderBottomWidth: 0 } : undefined
    ];

    return (
      <View style={containerStyles}>
        <StyledText style={styles.label}>{this.props.name}</StyledText>
        <StyledText style={styles.value} numberOfLines={1}>{this.props.value}</StyledText>
      </View>
    );
  }
}

SettingsAttribute.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  isLastItem: PropTypes.bool
};
