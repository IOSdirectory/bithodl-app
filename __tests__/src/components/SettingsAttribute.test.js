import React from 'react';
import renderer from 'react-test-renderer';
import SettingsAttribute from '../../../src/components/SettingsAttribute';

describe('SettingsAttribute', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SettingsAttribute
        name='8e69990a-32bf-4daf-9a40-2be6c45dad18'
        value='8ed9d949-9a68-4a3d-b69d-0856a632dacc'
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly as the last item', () => {
    const tree = renderer.create(
      <SettingsAttribute
        name='8e69990a-32bf-4daf-9a40-2be6c45dad18'
        value='8ed9d949-9a68-4a3d-b69d-0856a632dacc'
        isLastItem={true}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
