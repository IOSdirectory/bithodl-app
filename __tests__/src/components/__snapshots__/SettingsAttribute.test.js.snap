// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsAttribute renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#DDDDDF",
        "borderBottomWidth": 0.5,
        "marginLeft": 15,
        "paddingBottom": 12,
        "paddingRight": 15,
        "paddingTop": 12,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "fontWeight": "600",
        },
      ]
    }
  >
    8e69990a-32bf-4daf-9a40-2be6c45dad18
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "color": "#8F8E94",
          "position": "absolute",
          "right": 15,
          "textAlign": "right",
          "top": 12,
          "width": 150,
        },
      ]
    }
  >
    8ed9d949-9a68-4a3d-b69d-0856a632dacc
  </Text>
</View>
`;

exports[`SettingsAttribute renders correctly as the last item 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#DDDDDF",
        "borderBottomWidth": 0.5,
        "marginLeft": 15,
        "paddingBottom": 12,
        "paddingRight": 15,
        "paddingTop": 12,
      },
      Object {
        "borderBottomWidth": 0,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "fontWeight": "600",
        },
      ]
    }
  >
    8e69990a-32bf-4daf-9a40-2be6c45dad18
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "color": "#8F8E94",
          "position": "absolute",
          "right": 15,
          "textAlign": "right",
          "top": 12,
          "width": 150,
        },
      ]
    }
  >
    8ed9d949-9a68-4a3d-b69d-0856a632dacc
  </Text>
</View>
`;
